### ISSUE TITLE
Question: Deserialize search JSON to a SearchDescriptor

### ISSUE THREAD

**Issue by raghur** [elastic/elasticsearch-net#667]
Hi,
_Elasticsearch .net client newbie here_

Given a JSON search string, I don't see a way to construct a SearchDescriptor out of it. The QueryRaw and FilterRaw methods are there - but in my case, I could have a arbitrarily complex, fully formed search string. I'd like to use that to create the searchDescriptor, override a few params (esp size and from) and then reissue the search for reindexing.

Basically, looking for `SearchDescriptor<T>(string json)` or `nestClient.Deserialize<SearchDescriptor<T>>(string json)`

I'm aware that if I have a fully formed search string, I can use the lower level `client.Raw.Search` method but that means I have to handle all the deserialization.

Thanks.
